app/assets/javascripts/preferences/PreferencesViewWrapper.tsx

```diff
@@ -18,7 +18,8 @@ export const PreferencesViewWrapper: FunctionComponent<PreferencesViewWrapperPro
       <PreferencesView
         closePreferences={() => appState.preferences.closePreferences()}
         application={application}
-        mfaGateway={application}
+        mfaProvider={application}
+        userProvider={application}
```

Comment:
We use 2 separate types for `mfaProvider` and `userProvider` but pass the same `application` object to them. Probably it's better to either use one prop for `application` with appropriate name or pass application's respective fields to `mfaProvider` and `userProvider` (e.g., pass only `application.getUser` to `userProvider`)

Reply:
Separation of concerns. We can test  components that only depend on x:getUser(), that only depend on MfaProvider stuff, without mocking the whole WebApplication along with the kitchen sink :D

Reply:
It looks very strange tbh :D But I think we'll have better (practical) understanding of pros and cons of this approach only after writing some tests :)

Reply:
They're coming! :D 
Can I get an approval now? :)